[Backend] 루이 - 카드 이동 기능 구현

Louie-03 opened the pull request:
## Issue
- close #10 

## Overview (Required)
- 카드 이동 기능 구현입니다.
  - 카드 이동 시 updateDateTime을 갱신하도록 구현했습니다.
  - ActivityLogRepository.save()에서 만약 Action이 이동한다면 before_status를 넣어주도록 동적 쿼리를 구현했습니다.
- chore : ActivityLogRepository.findAll 메서드명 변경
  - create_datetime을 기준으로 정렬하기 떄문에 기능에 맞게 findAllOrderByCreateDatetimeDesc로 메서드명을 변경했습니다.

Louie-03 committed:
feat : 카드 이동 기능 구현-bn-tw-2020-#10

카드 이동 시 updateDateTime을 갱신하도록 구현했습니다.
ActivityLogRepository.save()에서 만약 Action이 이동한다면
before_status를 넣어주도록 동적 쿼리를 구현했습니다.

Louie-03 committed:
chore : ActivityLogRepository.findAll 메서드명 변경

create_datetime을 기준으로 정렬하기 떄문에 기능에 맞게 findAllOrderByCreateDatetimeDesc로 메서드명을 변경했습니다.

hwicode commented on BackEnd/src/main/java/todo/list/service/ActivityLogService.java:
```diff
@@ -17,8 +19,9 @@ public ActivityLogService(ActivityLogRepository activityLogRepository) {
+    @Transactional(readOnly = true)
     public List<ActivityLogResponse> findActivityLogs() {
-        List<ActivityLog> activityLogs = activityLogRepository.findAll();
+        List<ActivityLog> activityLogs = activityLogRepository.findAllOrderByCreateDatetimeDesc();
```
제 생각에는 이 부분은 service 계층이라서 좀 더 추상화된 메소드 이름이 더 괜찮은 것 같습니다!

Louie-03 committed:
chore : CardRepository.findAllSameStatus 메서드명 변경

update_datetime을 기준으로 정렬하기 떄문에 기능에 맞게
findSameStatusOrderByUpdateDatetimeDesc로 메서드명을 변경했습니다.

hwicode commented:
매우 깔끔하네요!! 수고많으셨습니다 루이!
